File: app/src/main/java/com/example/realm_template/MyApplication.java
package com.example.realm_template;

import android.app.Application;
import android.support.annotation.VisibleForTesting;

import com.example.realm_template.prngfix.PRNGFixes;
import com.squareup.leakcanary.LeakCanary;
import com.squareup.leakcanary.RefWatcher;

public class MyApplication extends Application {
    private static MyApplication self;
    public static MyApplication getInstance() {
        return self;
    }

    @VisibleForTesting
    public ApplicationComponent component;
    private RefWatcher refWatcher;

    public ApplicationComponent getComponent() {
        return component;
    }

    public RefWatcher getRefWatcher() {
        return refWatcher;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        self = this;

        PRNGFixes.apply();

        component = DaggerApplicationComponent.create();
        refWatcher = LeakCanary.install(this);
    }

}

File: kotlin_app/src/main/kotlin/com/example/realm_kotlin_template/MyApplication.kt
package com.example.realm_kotlin_template

import android.app.Application
import android.support.annotation.VisibleForTesting

import com.example.realm_kotlin_template.prngfix.PRNGFixes
import com.squareup.leakcanary.LeakCanary
import com.squareup.leakcanary.RefWatcher

import io.realm.Realm

open class MyApplication : Application() {

    @VisibleForTesting
    var component: ApplicationComponent? = null
    var refWatcher: RefWatcher? = null
        private set

    override fun onCreate() {
        super.onCreate()
        instance = this

        PRNGFixes.apply()

        Realm.init(this)

        component = DaggerApplicationComponent.create()
        refWatcher = LeakCanary.install(this)
    }

    companion object {
        var instance: MyApplication? = null
            private set
    }

}
